Packs/Nexthink/Integrations/Nexthink/Nexthink.py

```diff
@@ -0,0 +1,187 @@
+SEARCH_DEVICE_USING_DEVICE = "(select (*) (from device (where device ( eq name (string {0})))))".format(device)
+SEARCH_COMPLIANCE_PACKAGE_DEVICE = """(select ((device (*)) (package (*))) (from (device package)
+(with package (where package (eq name (pattern '*{0}*')))
+(where device (eq name (pattern '{1}')))))
+(limit 100))""".format(package, device)
+TEST_MODULE = "(select (name) (from device ) (limit 1))"
+
+
+def nexthink_request(method, nxql):
+    params = demisto.params()
+    username = params.get('credentials').get('identifier')
+    password = params.get('credentials').get('password')
+    base_url = params.get('url')
+    port = params.get('port')
+    verify_ssl = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+
+    if proxy:
+        proxies = handle_proxy()
+    else:
+        proxies = {
+            "http": None,
+            "https": None,
+        }
+
+    BASE_URL = 'https://{0}:{1}/2/query?platform=windows&format=json&query='.format(base_url, port)
+    NXQL = urllib.parse.quote(nxql)
+    urlFragment = BASE_URL + NXQL
+
+    try:
+        if method == 'POST':
+            response = requests.post(urlFragment, auth=(username, password), verify=verify_ssl, proxies=proxies)
+        else:
+            response = requests.get(urlFragment, auth=(username, password), verify=verify_ssl, proxies=proxies)
+
+        if response.status_code == 200:
+            return json.loads(response.content)
+        else:
+            return str(response.status_code)
+    except requests.Timeout:
+        raise
+    except requests.ConnectionError:
+        raise
+
+
+def nexthink_endpoint_details(device: None, ip: None):
+    if not device:
+        data = nexthink_request('GET', SEARCH_DEVICE_USING_IP)
+    else:
+        data = nexthink_request('GET', SEARCH_DEVICE_USING_DEVICE)
+    if len(data) > 0:
+        deviceEntry['Endpoint Name'] = data[0]['name']
+        deviceEntry['Last Logged On User'] = data[0]['last_logged_on_user']
+        deviceEntry['IP Address'] = data[0]['ip_addresses'][0]
+        deviceEntry['MAC Address'] = data[0]['mac_addresses'][0]
+        deviceList.append(deviceEntry)
+
+        dArgs = CommandResults(
+            outputs_prefix="Nexthink.Endpoint",
+            outputs=deviceList,
+            readable_output=tableToMarkdown('Nexthink Endpoint Details: ', deviceList),
+            raw_response=deviceList
+        )
+
+        return dArgs
+    else:
+        return 'Endpoint Not Found'
+
+
+def nexthink_installed_packages(device: None, package: None):
+    data = nexthink_request('GET', SEARCH_COMPLIANCE_PACKAGE_DEVICE)
+
+    if len(data) > 0:
+        for t in data:
+            entries = {}
+            entries['Package Name'] = t['package/name']
+            entries['Package Publisher'] = t['package/publisher']
+            entries['Package Version'] = t['package/version']
```

Comment:
what would happen if `t` will not include those keys? Is there a chance that the command will fail?

Reply:
it will be there i not seen such case so far